brave/brave-core
# Added URL Sanitizer service and Copy Clean URL item for links

Resolves https://github.com/brave/brave-browser/issues/23315

MVP implementation for url sanitization, spec https://docs.google.com/document/d/1ea7eF3s0WTsTlmMPr8LYp5KUyNg8eK4omjizo27qvJE/edit#

- Added a service that removes url parameters from query according to the rules shipped in resources.
- Added an item to the sharing hub menu to copy clean url
<img width="465" alt="image" src="https://user-images.githubusercontent.com/2965009/186643614-1c7a9715-450e-4761-854b-6c29d5c1568c.png">
- Added an item to the address bar context menu to copy clean url
<img width="557" alt="image" src="https://user-images.githubusercontent.com/2965009/186643577-55739fb2-03bc-4407-bd4d-7a79ae086458.png">

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Go to any url that meet config's rules and copy clean link.
- Check copied link should have removed parameters accoiring to the rules.
